src/CodeGenerator/src/Generator/ClientGenerator.php

```diff
@@ -67,12 +61,17 @@ public function getVersion() {
             $signatureVersions = \array_intersect($supportedVersions, $config['signVersions']);
             rsort($signatureVersions);
 
-            return strtr(sprintf("        return %s;\n", \var_export([
-                'endpoint' => $config['endpoint'],
-                'signRegion' => $config['signRegion'] ?? '%region%',
-                'signService' => $config['signService'],
-                'signVersions' => \array_values($signatureVersions),
-            ], true)), ['\'%region%\'' => '$region']);
+            return strtr('        return [
+                "endpoint" => "ENDPOINT",
+                "signRegion" => REGION,
+                "signService" => SIGN_SERVICE,
+                "signVersions" => SIGN_VERSIONS,
+            ];' . "\n", [
+                'ENDPOINT' => strtr($config['endpoint'], ['%region%' => '$region']),
+                'REGION' => isset($config['signRegion']) ? \var_export($config['signRegion'], true) : '$region',
+                'SIGN_SERVICE' => var_export($config['signService'], true),
+                'SIGN_VERSIONS' => \json_encode($signatureVersions),
```

Comment:
Haha. This is an ugly trick. I thought about it on #605, but I wasn’t sure it was pretty enough. :)

I’m okey with it :)

Reply:
hey hey